Source/WebCore/platform/SharedBuffer.cpp

```diff
@@ -121,6 +158,26 @@ Ref<SharedBuffer> FragmentedSharedBuffer::makeContiguous() const
     return SharedBuffer::create(WTFMove(combinedData));
 }
 
+auto FragmentedSharedBuffer::toIPCData() const -> IPCData
+{
+    size_t bufferSize = size();
+    if (!bufferSize)
+        return { };
+
+    if (useUnixDomainSockets() || bufferSize < minimumPageSize) {
+        if (m_segments.isEmpty())
```

Comment:
We probably don't need this check, I think the code below will do the right thing.